File: lib/model/screenless/ambiguous_blik_payment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:flutter_tpay/model/paymentMethod/blik_alias.dart';
import 'package:flutter_tpay/model/screenless/ambiguous_alias.dart';

part 'ambiguous_blik_payment.g.dart';

/// Class responsible for storing information about ambiguous BLIK payment.
/// - [transactionId] - id of transaction received from ScreenlessBlikAmbiguousAlias result
/// - [blikAlias] - alias used to create BLIK one click payment that returned ScreenlessBlikAmbiguousAlias result
/// - [ambiguousAlias] - ambiguous alias selected by user from list provided by ScreenlessBlikAmbiguousAlias result
@JsonSerializable(explicitToJson: true)
class AmbiguousBLIKPayment {
  final String transactionId;
  final BlikAlias blikAlias;
  final AmbiguousAlias ambiguousAlias;

  AmbiguousBLIKPayment({
    required this.transactionId,
    required this.blikAlias,
    required this.ambiguousAlias
  });

  factory AmbiguousBLIKPayment.fromJson(Map<String, dynamic> json) => _$AmbiguousBLIKPaymentFromJson(json);
  Map<String, dynamic> toJson() => _$AmbiguousBLIKPaymentToJson(this);
}

File: android/src/main/kotlin/com/tpay/model/screenless/AmbiguousBLIKPayment.kt
package com.tpay.model.screenless

import com.tpay.sdk.api.models.BlikAlias
import org.json.JSONObject
import com.tpay.sdk.api.screenless.blik.*
import com.tpay.extension.*

data class AmbiguousBLIKPayment(
    val transactionId: String,
    val blikAlias: BlikAlias,
    val ambiguousBlikAlias: AmbiguousBLIKAlias
) {
    companion object {
        private const val TRANSACTION_ID = "transactionId"
        private const val BLIK_ALIAS = "blikAlias"
        private const val AMBIGUOUS_ALIAS = "ambiguousAlias"

        fun fromJson(json: String): AmbiguousBLIKPayment = JSONObject(json).run {
            AmbiguousBLIKPayment(
                transactionId = getString(TRANSACTION_ID),
                blikAlias = getJSONObject(BLIK_ALIAS).run { getBlikAlias() },
                ambiguousBlikAlias = AmbiguousBLIKAlias.fromJson(getJSONObject(AMBIGUOUS_ALIAS))
            )
        }
    }
}